crates/core/src/operations/update.rs

```diff
@@ -627,28 +625,20 @@ pub(crate) async fn request_update(
 
             target
         } else {
-            // Check if we actually have any connected peers at all
-            let has_connections = op_manager.ring.connection_manager.num_connections() > 0;
-
-            if has_connections {
-                // We have connections but no suitable peer for this contract
-                return Err(OpError::RingError(RingError::NoCachingPeers(*key)));
-            } else {
-                // We truly have no peers, handle locally
-                tracing::debug!(
-                    "UPDATE: No peer connections available, handling contract {} locally",
-                    key
-                );
+            // No remote peers available, handle locally
+            tracing::debug!(
+                "UPDATE: No remote peers available for contract {}, handling locally",
+                key
+            );
 
-                // If no other peers, we should be subscribed and handle locally
-                op_manager
-                    .ring
-                    .add_subscriber(key, sender.clone())
-                    .map_err(|_| RingError::NoCachingPeers(*key))?;
+            // Subscribe ourselves
```

Comment:
this makes no sense, why would we subscribe to ourselves? The only consumers of our updates in this case are client applications, and that flow is handled completely independently of this.